test/unit/puppetlabs/services/jruby/jruby_pool_test.clj

```diff
@@ -96,30 +96,63 @@
   (let [pool-size 2
         config        (jruby-testutils/jruby-puppet-config {:max-active-instances pool-size})
         profiler      jruby-testutils/default-profiler
-        pool-context  (create-pool-context config profiler nil)
+        pool-context  (jruby-core/create-pool-context
+                        config profiler jruby-testutils/default-shutdown-fn)
         err-msg       (re-pattern "Unable to borrow JRuby instance from pool")]
     (with-redefs [jruby-internal/create-pool-instance! (fn [_] (throw (IllegalStateException. "BORK!")))]
-                 (is (thrown? IllegalStateException (jruby-agents/prime-pool! (:pool-state pool-context) config profiler))))
+                 (is (thrown? IllegalStateException (jruby-agents/prime-pool! pool-context config profiler))))
     (testing "borrow and borrow-with-timeout both throw an exception if the pool failed to initialize"
       (is (thrown-with-msg? IllegalStateException
             err-msg
-            (borrow-from-pool pool-context)))
+            (jruby-core/borrow-from-pool pool-context)))
       (is (thrown-with-msg? IllegalStateException
             err-msg
-            (borrow-from-pool-with-timeout pool-context 120))))
+            (jruby-core/borrow-from-pool-with-timeout pool-context 120))))
     (testing "borrow and borrow-with-timeout both continue to throw exceptions on subsequent calls"
       (is (thrown-with-msg? IllegalStateException
           err-msg
-          (borrow-from-pool pool-context)))
+          (jruby-core/borrow-from-pool pool-context)))
       (is (thrown-with-msg? IllegalStateException
           err-msg
-          (borrow-from-pool-with-timeout pool-context 120))))))
+          (jruby-core/borrow-from-pool-with-timeout pool-context 120))))))
 
 (deftest test-default-pool-size
   (logutils/with-test-logging
     (let [config (jruby-testutils/jruby-puppet-config)
           profiler jruby-testutils/default-profiler
-          pool (create-pool-context config profiler nil)
+          pool (jruby-core/create-pool-context config profiler jruby-testutils/default-shutdown-fn)
           pool-state @(:pool-state pool)]
-      (is (= (core/default-pool-size (ks/num-cpus)) (:size pool-state))))))
+      (is (= (jruby-core/default-pool-size (ks/num-cpus)) (:size pool-state))))))
 
+(defn create-pool-context
+  [max-requests]
+  (let [config (-> (jruby-testutils/jruby-puppet-config {:max-active-instances 1})
+                   (assoc :max-requests-per-instance max-requests))
```

Comment:
No biggie but you could get rid of the `assoc` and just put this key-value into the hash argument to `jruby-puppet-config`